frontends/web/src/routes/account/info/signingconfiguration.tsx

```diff
@@ -15,129 +15,101 @@
  * limitations under the License.
  */
 
-import { Component } from 'react';
+import { ReactNode } from 'react';
+import { useTranslation } from 'react-i18next';
 import { route } from '../../../utils/route';
 import { getCanVerifyXPub, IAccount, TBitcoinSimple, TEthereumSimple, TSigningConfiguration, verifyXPub } from '../../../api/account';
 import { getScriptName, isBitcoinBased } from '../utils';
 import { CopyableInput } from '../../../components/copy/Copy';
 import { Button } from '../../../components/forms';
 import { QRCode } from '../../../components/qrcode/qrcode';
-import { translate, TranslateProps } from '../../../decorators/translate';
 import style from './info.module.css';
+import { useLoad } from '../../../hooks/api';
 
-interface ProvidedProps {
+type TProps = {
     account: IAccount;
     info: TSigningConfiguration;
     code: string;
     signingConfigIndex: number;
+    children: ReactNode;
 }
 
-interface State {
-    canVerifyExtendedPublicKey: boolean;
-}
-
-type Props = ProvidedProps & TranslateProps;
-
-class SigningConfiguration extends Component<Props, State> {
-
-  public readonly state: State = {
-    canVerifyExtendedPublicKey: false,
-  };
-
-  public componentDidMount() {
-    this.canVerifyExtendedPublicKeys();
-  }
+export const SigningConfiguration = ({ account, info, code, signingConfigIndex, children }: TProps) => {
+  const canVerifyExtendedPublicKey = useLoad(getCanVerifyXPub(code));
+  const { t } = useTranslation();
 
-  private canVerifyExtendedPublicKeys = () => {
-    getCanVerifyXPub(this.props.code).then(canVerifyExtendedPublicKey => {
-      this.setState({ canVerifyExtendedPublicKey });
-    });
+  const verifyExtendedPublicKey = (signingConfigIndex: number) => {
+    verifyXPub(code, signingConfigIndex);
   };
```

Comment:
nit: looks like it could be directly called in the onClick handler without having a function here.